storage/boltz/link_collection.go

```diff
@@ -153,46 +157,118 @@ func (collection *linkCollectionImpl) GetLinks(tx *bbolt.Tx, id string) []string
 	return nil
 }
 
-func (collection *linkCollectionImpl) link(tx *bbolt.Tx, fieldBucket *TypedBucket, id string, associatedId string) error {
-	if fieldBucket.SetListEntry(TypeString, []byte(associatedId)).Err != nil {
+func (collection *linkCollectionImpl) link(tx *bbolt.Tx, fieldBucket *TypedBucket, id []byte, associatedId []byte) error {
+	if fieldBucket.SetListEntry(TypeString, associatedId).Err != nil {
 		return fieldBucket.Err
 	}
-	return collection.linkOther(tx, []byte(id), []byte(associatedId))
+	return collection.otherField.AddLink(tx, associatedId, id)
 }
 
-func (collection *linkCollectionImpl) linkOther(tx *bbolt.Tx, id []byte, associatedId []byte) error {
-	otherBaseBucket := collection.otherField.GetStore().GetEntityBucket(tx, associatedId)
-	if otherBaseBucket == nil {
-		return errors.Errorf("can't link to unknown %v with id %v", collection.otherField.GetStore().GetEntityType(), string(associatedId))
-	}
-	otherFieldBucket := otherBaseBucket.GetOrCreatePath(collection.otherField.GetPath()...)
-	return otherFieldBucket.SetListEntry(TypeString, id).Err
-}
-
-func (collection *linkCollectionImpl) unlink(tx *bbolt.Tx, fieldBucket *TypedBucket, id string, associatedId string) error {
-	if fieldBucket.DeleteListEntry(TypeString, []byte(associatedId)).Err != nil {
+func (collection *linkCollectionImpl) unlink(tx *bbolt.Tx, fieldBucket *TypedBucket, id []byte, associatedId []byte) error {
+	if fieldBucket.DeleteListEntry(TypeString, associatedId).Err != nil {
 		return fieldBucket.Err
 	}
-	return collection.unlinkOther(tx, []byte(id), []byte(associatedId))
+	return collection.otherField.RemoveLink(tx, associatedId, id)
 }
 
 func (collection *linkCollectionImpl) unlinkCursor(tx *bbolt.Tx, cursor *bbolt.Cursor, id []byte, associatedId []byte) error {
 	if err := cursor.Delete(); err != nil {
 		return err
 	}
-	return collection.unlinkOther(tx, id, associatedId)
+	return collection.otherField.RemoveLink(tx, associatedId, id)
+}
+
```

Comment:
Add LinkedSetSymbol, which extracts out some of the link/unlink code that was above and also adds ability to link/unlink  compound keys